plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchMetadata.java

```diff
@@ -660,6 +673,133 @@ private static boolean isPassthroughQuery(OpenSearchTableHandle table)
         return table.type().equals(OpenSearchTableHandle.Type.QUERY);
     }
 
+    @Override
+    public Optional<ProjectionApplicationResult<ConnectorTableHandle>> applyProjection(
+            ConnectorSession session,
+            ConnectorTableHandle handle,
+            List<ConnectorExpression> projections,
+            Map<String, ColumnHandle> assignments)
+    {
+        if (!isProjectionPushdownEnabled(session)) {
+            return Optional.empty();
+        }
+//        if (projections.stream().map(ConnectorExpression::getType).anyMatch((type) -> type instanceof ArrayType)) {
+//            return Optional.empty();
+//        }
+        // Create projected column representations for supported sub expressions. Simple column references and chain of
+        // dereferences on a variable are supported right now.
+        Set<ConnectorExpression> projectedExpressions = projections.stream()
```

Comment:
is this all a copy of what we have in mongo? any OS specific you had to do change here?

Reply:
there is os specific change while creating column handle